revenue_app/utils.py

```diff
@@ -51,23 +49,11 @@ def get_organizer_sales():
 def filter_transactions(transactions=None, **kwargs):
     if transactions is None:
         transactions = get_transactions()
-    conditions = []
-    if kwargs.get('event_id'):
-        conditions.insert(
-            0,
-            transactions['event_id'] == kwargs.get('event_id').strip()
-        )
-    if kwargs.get('email'):
-        conditions.insert(
-            0,
-            transactions['email'] == kwargs.get('email').strip()
-        )
-    if kwargs.get('eventholder_user_id') or kwargs.get('organizer_id'):
-        conditions.insert(
-            0,
-            transactions['eventholder_user_id'] ==
-            (kwargs.get('eventholder_user_id') or kwargs.get('organizer_id')).strip()
-        )
+    conditions = [
+        transactions[key] == kwargs.get(key).strip()
+        for key in kwargs
+        if key in ['event_id', 'email', 'currency', 'eventholder_user_id'] and kwargs.get(key)
```

Comment:
What are you trying to verify with `and kwargs.get(key)`? That the content is not `None`? If that's the case I'd check `and kwargs.get(key) is not None`. If you're looking to exclude _all_ falsy values this is fine, though

Reply:
To exclude all falsy values. Most of the time, the each key is equals to `''`